Docs 5222 mmp domain v4.2 (#840)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/mmp-concept.adoc

include::_attributes.adoc[]
endif::[]

> The Mule Maven Plugin enables you to integrate the packaging, testing, and deployment of your Mule applications with your Maven lifecycle. The plugin enables you to deploy your Mule application to different deployment targets such as standalone runtimes, clustered instances, and CloudHub.

Reviewer: ```suggestion
The Maven plug-in to Mule enables you to integrate the packaging, testing, and deployment of your Mule applications with your Apache Maven lifecycle. With the plug-in, you can deploy your Mule application to different deployment targets, such as standalone runtimes, clustered instances, and CloudHub.
```

Reviewer: We need to avoid giving the impression that MuleSoft has a product called Mule Maven Plugin. Maven belongs to Apache, which is very rigorous about pursuing attribution to the foundation. So, I think we need to say that Mule uses a plug-in to integrate with Apache Maven operations.

Reviewer: GLOBALLY reword to clarify this; I will stop marking it.

> The Mule Maven Plugin supports both Community and Enterprise editions.

Reviewer: The only "edition" that our PMM product naming source lists is this one:

Official name: Anypoint Platform Private Cloud Edition
Alternative: Anypoint Platform PCE

Following this pattern, should "both Community and Enterprise editions" be "both Anypoint Platform Community and Enterprise Editions" or "both Anypoint Platform Community Edition and Anypoint Platform Enterprise Edition"?

== Deploy a Mule Application to a Standalone Mule Runtime

. Make sure you added the Mule Maven Plugin to your `pom.xml` file.

> . Inside the plugin element, add a configuration for your standalone deployment as shown below:

Reviewer: ```suggestion
. Inside the plug-in element, add a configuration for your standalone deployment:
```

include::mule-runtime:ROOT:page$mmp-concept.adoc[tag=skipParameterDescription]
|===

== Deploy a Domain

> The Mule Maven Plugin supports deploying domains to standalone Mule instances and also when using the Mule agent.

Reviewer: ```suggestion
The Maven plug-in for Mule runtime engine supports deploying domains to standalone Mule instances and also to instances using the Mule agent.
```
I don't know if the rewording "to instances using the Mule agent" is technically accurate. What we need is parallelism: "to Mule instances and also when using" are not parallel.

> To deploy a domain, use the same configuration and deployment steps you use when deploying an application. For example, to deploy a domain to a Standalone instance:

Reviewer 2: to deploy a domain to a Standalone instance > to deploy a domain to a standalone instance:

> To deploy a domain, use the same configuration and deployment steps you use when deploying an application. For example, to deploy a domain to a Standalone instance:

Reviewer: ```suggestion
To deploy a domain, use the same configuration and deployment steps you use when deploying an application. For example, to deploy a domain to a standalone instance:
```
...as Kevin has pointed out already.

> . Add the Mule Maven Plugin to your `pom.xml` file.

Reviewer: ```suggestion
. Add the Maven plug-in to Mule to your `pom.xml` file.
```
If this plug-in is referred to in syntax in some other way, be sure to use that syntax instead, enclosed in ``.

> . Inside the plugin element, add a configuration for your standalone deployment. For example:

Reviewer: ```suggestion
. Inside the plug-in element, add a configuration for your standalone deployment, replacing the placeholder values with your local Mule runtime engine information:
```

+
[source,xml,linenums]
----
<plugin>

  <configuration>
    <standaloneDeployment>
      <muleHome>${mule.home.test}</muleHome>
      <muleVersion>${app.runtime}</muleVersion>
    </standaloneDeployment>
  </configuration>

</plugin>
----
+

> The example is using placeholder values. Configure each value with your local Mule runtime engine information.

Reviewer: DELETE in favor of the edit to the example intro.

> . From the command line in your project's folder, package the domain and run the deploy goal:

Reviewer: I am not familiar with something called a "deploy goal"; is it actually something that you can "run"? Running a goal sounds incorrect to me; running a deployment sounds correct.

+
[source,console,linenums]
----
$ mvn clean package deploy -DmuleDeploy
----

> == Encrypting credentials

Reviewer: ```suggestion
== Encrypting Credentials
```

> NOTE: Make sure that username and password are not set in the deployment configuration, as they have precedence over the defined server id.

Reviewer: ```suggestion
NOTE: Make sure that the username and password are not set in the deployment configuration, because they have precedence over the defined server ID.
```
...or something like "...configuration or they will overwrite the defined server ID."

> == Mule Maven Plugin Goals Description

Reviewer: ```suggestion
== Maven Plug-in to Mule Goals
```

> The Mule Maven Plugin has three goals that are explained below: Package, Deploy and Undeploy.

Reviewer: ```suggestion
The Maven plug-in to Mule runtime engine has three goals: package, deploy, and undeploy.
```
Are these officially called "goals" or are we referring to purposes or operations? Should we refer to these by their noun forms: packing, deployment, and undeployment?

> Each goal accepts different parameters that change the plugin behavior. When you provide a parameter from the command line, you have to prepend `-D` to the parameter name.

Reviewer: ```suggestion
Each goal accepts parameters that are unique to the desired plug-in behavior, although each parameter must begin with `-D`.
```

For example, to execute the `package` goal and set the `attachMuleSources` parameter, you run the following command:

> [source,console,linenums]

Reviewer: Is this part of the intro or the command example?

----
mvn package -DattachMuleSources
----

=== Package Goal

This goal generates the application JAR file.

> Bind's by default to the http://maven.apache.org/ref/3.6.1/maven-core/lifecycles.html[lifecycle phase]: `package`

Reviewer: Is this supposed to be part of or a followup to the previous sentence, stating that the goal "binds" (not "bind's") to the lifecycle phase? What is the ": `package`" syntax? If it is at the end of the sentence, if should be followed by a period, but I am not sure that this is a complete sentence, as-is....

> .Optional parameters

Reviewer: ```suggestion
Optional Parameters
```

=== Deploy Goal

This goal uploads and deploys the application JAR file to any of the application deployment targets.

> Bind's by default to the http://maven.apache.org/ref/3.6.1/maven-core/lifecycles.html[lifecycle phase]: `deploy`

Reviewer: ditto previous questions about this sentence, in the "package" section

> .Optional parameters

Reviewer: ditto

[%header%autowidth.spread,cols="a,a,a"]
|===
|Parameter | Type | Description
|`artifact` | String | Path to the application JAR file to be deployed. By default is set to the application `target` folder.

> |`muleDeploy` | boolean | Instructs the plugin to deploy using the deployment strategy defined in the plugin configuration. If it is not set, the plugin uploads the artifacts to the repository defined in the `distributionManagement` section on the application's `pom.xml` file.

Reviewer: ```suggestion
|`muleDeploy` | boolean | Instructs the plug-in to deploy using the deployment strategy defined in the plug-in configuration. If it is not set, the plug-in uploads the artifacts to the repository defined in the `distributionManagement` section of the application's `pom.xml` file.
```
QUESTION: Does "if it is not set" refer to this parameter not being set or to the deployment strategy not being defined in the plug-in?

|===

=== Undeploy Goal

This goal removes an application from any of the application deployment targets.

> Uses the information from the plugin configuration to remove the application from the defined deployment targets.

Reviewer: ```suggestion
It uses the information from the plug-in configuration to remove the application from the defined deployment targets.
```
